### ISSUE TITLE
[TF2] Remove 2 GB+ of HL2 sounds

### ISSUE THREAD

**Issue by henke37** [ValveSoftware/Source-1-Games#2186]
Teamfortress 2 is a large game, taking up more than 10 GB of space. But a notable part of this is completely unused and unneeded content inherited from HalfLife 2.

In particular, all of the story and npn voice lines are still included. None of that is useful in TF2.

While at it, compress all the sounds inherited from HL2.

**Comment by hahagoni** [comment]
Isn't the HL2 part used in hammer though?

**Comment by henke37** [comment]
Hammer does generally not play sounds. And even if it did, far from every single sound from HL2 would be needed. Especially not stuff like story voice lines.

Basically, TF 2 treats HL 2 as a base game when it comes to resources. Instead of finding a minimum shared subset of resources, it brings in all of them. For example, the SFX for a respawning item is from HL2. It is those kinds of shared sounds that made it bring in the resources from HL2. But this much unused stuff? Too much. 

Also, I don't think this is a feature request. This is a bug.

**Comment by tom-GER** [comment]
Yup, these voicelines are not used in any way (except some custom community maps)

There is no need to keep them ingame

**Comment by OscarOrSomething** [comment]
+1

**Comment by 404UserNotFound** [comment]
Where exactly would the line be drawn on what gets removed? As a mapmaker and a plugin developer (average and still learning for both), I'd advise against the removal of content such as:

1. Generic voice lines that have nothing to do with HL2, as these can be used for custom maps and SourceMod plugins alike. Yes, I'm saying cut the main HL2 related voice lines that are long drawn out "cutscene" voice lines. But keep individual smaller word clips (i.e. your "No" and "Yes" clips and the like).
2. I'm going to lump several things into one entry because I can. Models/Textures/Particle effects/Sounds related to objects (i.e. physics sounds for things scraping on certain ground types). Yes, all of them. Leave them in. 99.9% of community made maps utilize HL2 assets. Removal of any of this content would really screw up a lot of maps. Oh, and again, they're useful for SourceMod plugins. Good example being MasterOfTheXP's TF2 Sandbox plugin which was basically Garry's Mod prop building in TF2.

Imagine having clients have to download HL2 models for plugins, or downloading maps with increased filed size due to having to pack the HL2 assets into the map. There is one thing that Valve could do to remedy the removal of HL2 "shared" content in TF2's files.

If total shared content removal is done, there would have to be some sort of fix for plugins and maps that use the HL2 assets where the game is able to locate the HL2 game install (because with custom install directories, someone out there could have TF2 and HL2 in two different directories) and then load and precache all that content.

Just my two cents. Sorry if it seems erratically written or worded weirdly or something, I'm high.

**Comment by SnowshoeIceboot** [comment]
Neither CS:GO or L4D1/L4D2 have these HL2 files, further reason to drop them from TF2.

**Comment by TheSqrtMinus1** [comment]
This would break TONS of custom maps and plugins on TONS of community servers.

Not gonna happen, sorry.

**Comment by henke37** [comment]
Actual numbers please. This isn't a matter of "if" it will break stuff. but a matter of what stuff should and should not break. Things such as "utility" and "popularity" needs to be factored in here. And remember, this isn't a simple matter of either removing everything or keeping everything. It's a matter of deciding which files (in the archives) to keep. Each file needs to be judged individually.

**Comment by SnowshoeIceboot** [comment]
Custom maps that are broken can be updated by putting the HL2 sounds in the pakfile.

**Comment by 404UserNotFound** [comment]
Before I post, I'll clarify one thing to prevent anyone from getting confused: In my responses below, I'm speaking as if the idea is to remove _**all**_ the files. I'm not worrying about "Each file needs to be judged individually." because that'll never happen, as I state below.

> Custom maps that are broken can be updated by putting the HL2 files in the pakfile, or you can edit the maps and plugins to use TF assets instead.
No. That just adds more filesize to the maps, and leaves countless maps broken. Countless more maps will remain broken. And the fact you say to switch plugins to TF2 assets is retarded. Spawn the Rollermine for example is the HL2 rollermine. There's no way to switch that to TF2 assets. Your idea is stupid.

> You can also play Gmod if you really need a ton of HL2 stuff in a Team Fortress game.
Not even worth an indepth response, but I will direct you to MasterOfTheXP's TF2 Sandbox plugin which uses every HL2 model. I've also got a TF2 sandbox plugin in the works (that I've commissioned) that will use all the models. Numerous other plugins ("Spawn a Rollermine" in TF2 for example) use the rollermine model and sounds. There's also pet plugins that use HL2 mobs for "pets" in TF2. Not everyone wants to play Garry's Mod. That's a lazy excuse.

> Actual numbers please.
Impossible to get numbers on how many maps or plugins would be broken, and you know that.

> This isn't a matter of "if" it will break stuff
Yes it is.

> but a matter of what stuff should and should not break.
No it isn't.

> Things such as "utility" and "popularity" needs to be factored in here.
[What the fuck?](http://2static2.fjcdn.com/thumbnails/comments/5965859+_47e2448450c082fc7c557228bb6e6e93.jpg)

> And remember, this isn't a simple matter of either removing everything or keeping everything. It's a matter of deciding which files (in the archives) to keep. Each file needs to be judged individually.
[You're serious?](https://www.youtube.com/watch?v=FopyRHHlt3M)

> Neither CS:GO or L4D1/L4D2 have these HL2 files, further reason to drop them from TF2.
Congratulations for those two games, but they should stay in TF2. The last thing the community needs is to worry about having to pak more assets into their maps (which jacks filesize up, no thanks), or having to issue file downloads of files that everyone should already have in HL2, under custom directories just to keep SourceMod plugins that use HL2 models in TF2 working. It's more hassle than it's worth.

**Comment by asherkin** [comment]
It is an excellent idea to strip the install size down to only what the base game actually requires - the legacy files could be offered as a "mod" in steamcmd to avoid breaking things (much like CS/DoD/HLMP are distributed for GoldSrc).

**Comment by 404UserNotFound** [comment]
Thank you for commenting, Asherkin. That was indeed something I wanted to suggest, but I didn't quite know how to explain it best, but you did a great job.

It would allow server owners to continue to run plugins and maps and the assets would be there for the server to use to keep said things from appearing broken....however, what about the connecting client? How would they see these legacy files if they aren't in that player's TF2 directory from the steamcmd mod? I believe this would result in the connecting client seeing ERROR models everywhere.

**Comment by 404UserNotFound** [comment]
FYI: TF2 has an entity called "gibshooter" that fires out HL2 skeletal gibs. I'm sure there's probably some other entities and even places in TF2's source code where HL2 sounds/models are used. If content changes are made, hopefully Valve remember to go through all of TF2's code and make sure any HL2 assets that are called on are properly accounted for.

**Comment by Fantasmos** [comment]
I have mixed feeling on this because there's too many existing maps that utilise HL2 assets.

I suggest that during the map-loading phase TF2 will identify assets that are apart of HL2, and download them onto the local install of TF2. Therefore all existing maps will work whilst improving the download size for a majority of the game.

**Comment by doublah** [comment]
Custom servers and maps can pack in any specific sounds they want, just like every other custom sound file in the game. This is just a waste of space being in the game.

**Comment by Chdata** [comment]
I rely a lot on various HL2 resources that happen to be packed into TF2 for designing custom sourcemod plugins, as do many others.

However, my servers are a few of thousands, and it's possible for me to add custom downloads.

We shouldn't be sacrificing the efficiency for tens of thousands of customers just for a few server owners.

Only things that both HL2 and TF2 actually share should be kept - anyone using anything else will see error messages when their stuff isn't precached and will be able to deal with it.

As for maps that use HL2 props in maps ... hopefully something like what @Fantasmos said is viable, because it tends to be unlikely that map makers will go back and revisit their maps - I've seen many map makers put out some maps and then disappear for years.

**Comment by tkart64** [comment]
I think the problem here is that what we see as a trend is that user content is being treated more like an addition to the base game like a user spray on the wall, and valve keeps treating user content like official valve content where user content has the same ground and ability to influence user experience as official content which is wrong imo, because 99 percent of user content is senseless bloatware that doesnt add anything meaningful to the game. some css and dods servers have completely absurd level of user generated bloatware with maps that have poor optimization, bugged entities and garbled textures and that is all being played because there is no adequate filtering system in the form of steam workshop for these games, i seriously doubt that users want to spend 5 minutes waiting for custom servers sounds and vip player models along with who knows what to download which will potentially also break the game or make it go as slow as crysis on a netbook. even duke nukem 3d has workshop but not css, dods and cs1.6. we need persistent looking gameplay across all valve games, be it cs1.6 or csgo.

**Comment by henke37** [comment]
Not, that is not this particular problem. This is just valve sucking at build management. Your issue is something else.

**Comment by tkart64** [comment]
> Not, that is not this particular problem. This is just valve sucking at build management. Your issue is something else.

I think the only reason why hl2 content was not removed from source based games is to keep compatibility with those badly designed user generated maps which is why these two things are basically the same problem of valve treating user content the same as their own. We don't need people with no mapping design skills or just basic workmanship to dictate how the game is going to be played for the rest of the player base. I have played on those servers with maps that look like something in between minecraft and hl2 and css and tf2, and trust me if those user maps had at least gone through some testing they all would have been banned because it could take no more than a week of effort to make them look presentable, but for some reason it's easier to just shove it to the players and move on to making another half-finished map.

**Comment by tkart64** [comment]
I understand that the issue of user content is not the same as topic above, it just baffles me how valve could do this kind of stuff for the sake of some modders over majority of playerbase.

**Comment by 404UNFca** [comment]
> 
> 
> The thing is, that doesn't make sense. This is junk they accidentally added in the steampipe conversion. It wasn't there at launch. This is clearly incompetence, not compatibility.
> 
> You seem to be projecting a lot of issues here. Custom content is what keeps games alive. But that's not this issue. This is nothing more than incompetent packaging.
> 
> Valve could do a lot to improve the game. But this issue is not your issue. This is not the place for your complaints. They are valid, but they are off topic to this issue.

Fairly certain the HL2 content has always been in TF2's files from the get-go.

**Comment by 404UNFca** [comment]
> The thing is, that doesn't make sense. This is junk they accidentally added in the steampipe conversion. It wasn't there at launch. This is clearly incompetence, not compatibility.

Fairly certain the HL2 content has always been in TF2's files from the get-go. I seem to recall the shared content being in the files as far back as 2009 which is long before Steampipe.

**Comment by tkart64** [comment]
Yes it was part of all source games css, tf2 and dods, but why would it be needed for official content that has separate content set for each source game, aside from as mentioned above to maintain compatibility with badly designed user maps that use hl2 content in other source games, which mind you is not a pretty site to see. It has more to do with company policy of supporting user content than sound design idea that each game has its own separate content set which is clearly a more well thought out idea rather than have completely garbled policy of supporting all user content no matter how badly designed it is and have hl2 maps in dods tf2 and css even when it makes no sense whatsoever, and not only that but a policy of not implementing steam workshop in css and dods for the same reason that it will cause those badly deisned maps to get downvoted by players or loose compatibility, even though you can clearly see how badly designed it all is.

**Comment by ldesgoui** [comment]
If any work is planned by Valve in this regard, I'd like to point attention to the fact that the TF2 Dedicated Server works completely fine with something like 2.6GB (probably less, even) out of the 7.6GB total; this can be achieved by filtering out the default maps packaged with the game.
Those maps (and honestly, any non-core asset) could be hosted on the workshop and downloaded whenever needed.
Tighter integration with the workshop could
- reduce the requirement for server owners to host fastDL servers, 
- reduce random asset bloat, 
- let users unsubscribe to assets they don't plan on using anymore, 
- help the dev team to push map updates without requiring a game package update, 
- as well as (I assume) helping the contribution loop that happens when third party mapmakers get their maps in the official game.
I can see a lot of potential in the workshop integration, but it's currently buggy with demos, lacking multiple features, etc.

**Comment by tkart64** [comment]
It will be really nice if valve decides to cleanup and update game content both user and official to save up some space so that people waste less space downloading broken or duplicate content, i think it saves a lot of time especially since most games are internet based, when i choose which games i play i always pay attention how much space it takes on my ssd as i have limited space.

**Comment by SirYodaJedi** [comment]
I don't think this is a good idea, due to how much it breaks; rather, it would be better for Source 2013 Multiplayer games to have a shared folder for the base HL2 assets, like they did back when they used GCF instead of VPK. See #708.

**Comment by henke37** [comment]
ADPCM is a legacy codec. There is little point in using such an outdated codec.

**Comment by SirYodaJedi** [comment]
> ADPCM is a legacy codec. There is little point in using such an outdated codec.

And Source 1 is a legacy engine. Using ADPCM ensures that the WAV files used for VO remain as WAV files, as that is what maps and soundscripts that use those files are looking for.
